prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -471,22 +470,9 @@ int rxr_pkt_init_eor(struct rxr_ep *ep, struct rxr_rx_entry *rx_entry, struct rx
 void rxr_pkt_handle_eor_send_completion(struct rxr_ep *ep,
 					struct rxr_pkt_entry *pkt_entry)
 {
-	struct rxr_eor_hdr *eor_hdr;
 	struct rxr_rx_entry *rx_entry;
 
-	eor_hdr = (struct rxr_eor_hdr *)pkt_entry->pkt;
-
-	rx_entry = ofi_bufpool_get_ibuf(ep->rx_entry_pool, eor_hdr->rx_id);
-	assert(rx_entry && rx_entry->rx_id == eor_hdr->rx_id);
-
-	if (rx_entry->op == ofi_op_msg || rx_entry->op == ofi_op_tagged) {
-		rxr_cq_write_rx_completion(ep, rx_entry);
-	} else {
-		assert(rx_entry->op == ofi_op_write);
-		if (rx_entry->cq_entry.flags & FI_REMOTE_CQ_DATA)
-			rxr_cq_write_rx_completion(ep, rx_entry);
-	}
-
+	rx_entry = pkt_entry->x_entry;
 	rxr_release_rx_entry(ep, rx_entry);
```

Comment:
Do we need to hold on to the rx_entry till we get the EOR Tx completion? We can release it soon after writing a Rx completion to the application.

Reply:
Just  after I wrote that comment, I realized we have state about the EOR pkt in the rx_entry, so ignore what I said.

Reply:
Yeah. The send of EOR can fail due to EAGAIN, in that case ```rxt_pkt_post_or_queue``` will queue ```rx_entry``` and return success, but we cannot release rx_entry in that case. We can move the release of ```rx_entry``` to ```handle_eor_sent``` instead of ```handle_eor_send_completion```.

Reply:
Yeah, I missed that when I first commented. What you have here works better, and releasing after send completion makes sense.